integration/container_limits_test.go

```diff
@@ -0,0 +1,265 @@
+	sbConfig := PodSandboxConfig("sandbox1", "stats")
+	sb, err := runtimeService.RunPodSandbox(sbConfig, *runtimeHandler)
+	require.NoError(t, err)
+	defer func() {
+		assert.NoError(t, runtimeService.StopPodSandbox(sb))
+		assert.NoError(t, runtimeService.RemovePodSandbox(sb))
+	}()
+
+	testImage := GetImage(ResourceConsumer)
+	img, err := imageService.PullImage(&runtime.ImageSpec{Image: testImage}, nil, sbConfig)
+	require.NoError(t, err)
+	defer func() {
+		assert.NoError(t, imageService.RemoveImage(&runtime.ImageSpec{Image: img}))
+	}()
+
+	// consume 200 MB memory for 60 seconds.
+	var resourceLimits, command ContainerOpts
+	if goruntime.GOOS == "windows" {
+		// -d: Leak and touch memory in specified MBs
+		// -c: Count of number of objects to allocate
+		command = WithCommand("testlimit.exe", "-accepteula", "-d", "1", "-c", "200")
+		resourceLimits = WithWindowsResources(&runtime.WindowsContainerResources{
+			MemoryLimitInBytes: 100 * 1024 * 1024,
+		})
+	} else {
+		// -m: spawn N workers spinning on malloc()/free()
+		// --vm-bytes: malloc B bytes per vm worker (default is 256MB)
+		command = WithCommand("stress", "-m", "200", "--vm-bytes", "1M", "--vm-hang", "0", "-t", "60")
+		resourceLimits = WithResources(&runtime.LinuxContainerResources{
+			MemoryLimitInBytes: 100 * 1024 * 1024,
+		})
+	}
+
+	t.Logf("Create a container config and run container in a pod")
+	containerConfig := ContainerConfig(
+		"container1",
+		testImage,
+		command,
+		resourceLimits,
+		WithTestLabels(),
+		WithTestAnnotations(),
+	)
+	cn, err := runtimeService.CreateContainer(sb, containerConfig, sbConfig)
+	require.NoError(t, err)
+	defer func() {
+		assert.NoError(t, runtimeService.RemoveContainer(cn))
+	}()
+	require.NoError(t, runtimeService.StartContainer(cn))
+	defer func() {
+		assert.NoError(t, runtimeService.StopContainer(cn, 10))
+	}()
+
+	statChecker := memStatChecker(99, 101, cn)
+	require.NoError(t, Eventually(statChecker, time.Second, 30*time.Second))
+	require.NoError(t, Consistently(statChecker, time.Second, 30*time.Second))
+}
+
+// Test to verify that the memory limit is enforced.
+func TestContainerUpdateMemoryLimit(t *testing.T) {
+	if goruntime.GOOS == "windows" {
+		t.Skipf("Not supported on Windows.")
+	}
+	t.Logf("Create a pod config and run sandbox container")
+	sbConfig := PodSandboxConfig("sandbox1", "stats")
+	sb, err := runtimeService.RunPodSandbox(sbConfig, *runtimeHandler)
+	require.NoError(t, err)
+	defer func() {
+		assert.NoError(t, runtimeService.StopPodSandbox(sb))
+		assert.NoError(t, runtimeService.RemovePodSandbox(sb))
+	}()
+
+	testImage := GetImage(ResourceConsumer)
+	img, err := imageService.PullImage(&runtime.ImageSpec{Image: testImage}, nil, sbConfig)
+	require.NoError(t, err)
+	defer func() {
+		assert.NoError(t, imageService.RemoveImage(&runtime.ImageSpec{Image: img}))
+	}()
+
+	// consume 200 MB memory for 120 seconds.
+	var resourceLimits, command ContainerOpts
+	if goruntime.GOOS == "windows" {
```

Comment:
not consistent with `t.Skipf("Not supported on Windows.")` in L95

Reply:
Indeed. This test is meant to also run on Windows when support for updating running containers is added. I suppose I can remove it until then.